sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportStatus.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * Represents the status of the viewport.
+ *
+ * It could be either a [ViewportState] or [ViewportTransition].
+ */
+sealed class ViewportStatus {
+  /**
+   * Represents the current status is a [ViewportState].
+   *
+   * The state is null if current status is IDLE.
+   */
+  class State(val state: ViewportState?) : ViewportStatus() {
+    override fun getTargetState(): ViewportState? {
+      return state
+    }
+  }
+
+  /**
+   * Represents the current status is a [ViewportTransition].
+   */
+  class Transition(
```

Comment:
Same as above.

Reply:
Added overrides for equals and hash for ViewportStatus class, which takes the Status and Transition into consideration already.